src/Time/Actions/SelfStartActions.hpp

```diff
@@ -0,0 +1,396 @@
+    const TimeDelta initial_step = db::get<::Tags::TimeStep>(box);
+    // The slab number increments each time a new point is generated
+    // until it reaches zero.
+    const auto values_needed =
+        -db::get<::Tags::Next<::Tags::TimeId>>(box).slab_number();
+    const TimeDelta self_start_step = initial_step / (values_needed + 1);
+
+    db::mutate<::Tags::TimeStep>(
+        make_not_null(&box), [&self_start_step](
+                                 const gsl::not_null<
+                                     db::item_type<::Tags::TimeStep>*>
+                                     time_step) noexcept {
+          *time_step = self_start_step;
+        });
+
+    return std::make_tuple(
+        db::create_from<db::RemoveTags<>,
+                        db::AddSimpleTags<Tags::InitialValue<::Tags::TimeStep>,
+                                          Tags::InitialValue<variables_tag>>>(
+            std::move(box), std::make_tuple(initial_step),
+            std::make_tuple(db::get<variables_tag>(box))));
+  }
+};
+
+/// \ingroup ActionsGroup
+/// \ingroup TimeGroup
+/// Terminates the self-start phase if the required order has been
+/// reached.
+///
+/// Uses:
+/// - ConstGlobalCache: nothing
+/// - DataBox: Tags::Next<Tags::TimeId>
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies: nothing
+template <typename ExitTag>
+struct CheckForCompletion {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTags>& box,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    // The self start procedure begins with slab number
+    // -number_of_past_steps and counts up.  When we reach 0 we should
+    // start the evolution proper.  The first thing the evolution loop
+    // will do is update the time, so here we need to check if the
+    // next time should be the first real step.
+    return std::tuple<db::DataBox<DbTags>&&, bool, size_t>(
+        std::move(box), false,
+        db::get<::Tags::Next<::Tags::TimeId>>(box).slab_number() == 0
+            ? tmpl::index_of<ActionList, ::Actions::Label<ExitTag>>::value
+            : tmpl::index_of<ActionList, CheckForCompletion>::value + 1);
+    // Once we have full support for phases this action should
+    // terminate the phase:
+    // return std::tuple<db::DataBox<DbTags>&&, bool>(
+    //     std::move(box),
+    //     db::get<::Tags::Next<::Tags::TimeId>>(box).slab_number() == 0);
+  }
+};
+
+/// \ingroup ActionsGroup
+/// \ingroup TimeGroup
+/// Set the next time to the start time and increment the slab number
```

Comment:
This may be that I'm just not having the best luck with thinking right now, but I'm unclear as to whether the time is set to the start time each time the action is called or only if an order increase is to be done.